src/pages/dashboard/_StateCumulativeTestsContainer.js

```diff
@@ -104,7 +154,15 @@ export default function CumulativeTestsByStateContainer() {
     }
   `)
 
-  const data = groupAndSortStateDaily(query)
+  const [useTestsPerCapita, setUseTestsPerCapita] = useState(false)
+
+  const toggleChartData = () => setUseTestsPerCapita(u => !u)
+
+  const allData = groupAndSortStateDaily(query)
```

Comment:
this is performed on every render even though the value should never change. This one is a good candidate for `useMemo`.

Because the per cap toggle is the only thing that should trigger a rerender (only mutation to state in this component) the `useMemo` below doesn't accomplish anything. (not to mention you have to pass an array of dependencies for `useMemo` to work properly).

So I'd memoize `allData` and not `data` - although it doesn't hurt to memoize data too.

Reply:
Ah, yep. Memoized (and left `data` as is) in 14e451f.